### ISSUE TITLE
Doorbell support

### ISSUE THREAD

**Issue by pietro4711** [peterbaumert/ioBroker.unifi-protect#36]
Es wäre grandios, wenn auch die Doorbell samt dem Setzen von Messages supported werden würde !
Siehe z.B. hier: [https://github.com/hjdhjd/homebridge-unifi-protect/blob/master/docs/Doorbell.md](url)

**Comment by fladdie** [comment]
+10 ;-)
Das wäre wirklich absolut Grandios!

**Comment by fladdie** [comment]
@pietro4711 ,

ich hab mich mal aufgrund deines Links daran versucht mittels HomeBridge Adapter das von dir verlinkte Homebridge Addon zum laufen zu bekommen. Klappt auch soweit. Leider bekomme ich einige Sachen nicht zum Laufen. So bekomme ich leider nirgends mit wenn die Klingel betätigt wird. Auch die MQTT einbindung ist irgendwie nur so halb funktional.... Also entweder übersehe ich etwas oder das Teil läuft leider nicht so richtig unter ioBroker.

**Comment by pietro4711** [comment]
wie hast Du denn den Homebridge-Adapter installiert ? Ich verwende in IOBroker den "Homebridge Zubehör Manager" - dieser läuft im Wrapper-Modus und hier habe ich das oben genannte Homebridge installiert...

**Comment by fladdie** [comment]
Im Prinzip genauso wie du gerade geschrieben hast. Mit homebridge-adapter meinte ich den hombridge Zubehör Adapter,ebenfalls im wrapper Modus. Aber wie zuletzt schon geschrieben bekomme ich hier leider keine Klingelevents mit und auch per mqtt bekomme ich den Klingel Button nicht zu sehen. Am 24.06.2021 22:26 schrieb pietro4711 ***@***.***>:

wie hast Du denn den Homebridge-Adapter installiert ? Ich verwende in IOBroker den "Homebridge Zubehör Manager" - dieser läuft im Wrapper-Modus und hier habe ich das oben genannte Homebridge installiert...

**Comment by Xenon-s** [comment]
@peterbaumert wie ist der Stand hier? Kann ich irgendwie unterstützen? Ich hätte das Teil schon gerne unterstützt..

**Comment by peterbaumert** [comment]
> @peterbaumert wie ist der Stand hier? Kann ich irgendwie unterstützen? Ich hätte das Teil schon gerne unterstützt..

Hi, sorry ich war die letzten Monate ziemlich unter Wasser. Ja also wenn du mal Zeit hast für ne kurze TeamViewer/wasauchimmerremotesoftware Session hast. Bin im iobroker discord Pete#9330

**Comment by Xenon-s** [comment]
@peterbaumert kein Problem. Ich habe im forum auch schon gefragt, ist es eigentlich nicht möglich das ganze anstatt per API Abfrage mit websocket zu bauen? Ich wollte das am Wochenende mal testen, da bei 1 Sekunde Abfrageintervall oft meine Daten nicht mehr aktualisiert werden.

Ich melde mich über discord bei dir

**Comment by peterbaumert** [comment]
An das websocket thema habe ich mich gerade ran gesetzt. Das war ganz oben auf der Liste wenn ich wieder zeit habe :)

**Comment by Xenon-s** [comment]
Hatte darüber schon kurz mit dutch gesprochen. Ich habe Samstag den ganzen Tag Zeit, falls du Lust und Zeit hast😉

**Comment by peterbaumert** [comment]
![realTimeStuff](https://user-images.githubusercontent.com/1212934/135751261-53f8f857-b262-4a64-8e94-b71831721272.png)
Also lastMotion funktionier schonmal definitiv. den lastRing und lcdMessage kram konnte ich mangels hardware net testen. würde mich da über ne rückmeldung des raw teils freuen :)

**Comment by pietro4711** [comment]
Schaut auf den ersten Blick sehr gut aus !! Auch lastRing mit der Doorbell klappt tadellos !
Wie soll denn das mit dem raw-Wert funktionieren ? Einfach den Text reinschreiben hat bei mir nicht geklappt.

**Comment by Xenon-s** [comment]
@peterbaumert ja die funktionieren, aber durch das häufige API abrufen bekomme ich nach einigen Stunden keine Daten mehr rein. Deswegen fragte ich ja nach dem websocket.

**Comment by peterbaumert** [comment]
> @peterbaumert ja die funktionieren, aber durch das häufige API abrufen bekomme ich nach einigen Stunden keine Daten mehr rein. Deswegen fragte ich ja nach dem websocket. 

Was genau meinst du? Das hier nutzt den webhook der aktiv daten sendet. Da wird keine api abgefragt. Du solltest dein intervall erhöhen.

**Comment by peterbaumert** [comment]
> Schaut auf den ersten Blick sehr gut aus !! Auch lastRing mit der Doorbell klappt tadellos !
> 
> Wie soll denn das mit dem raw-Wert funktionieren ? Einfach den Text reinschreiben hat bei mir nicht geklappt.
> 
> 

Ne da wird einfach genau das reingeschrieben was von der api kommt. Und das könntest du mir hier her kopieren dann kann ich direkt states aus den werten machen.

**Comment by Xenon-s** [comment]
> > @peterbaumert ja die funktionieren, aber durch das häufige API abrufen bekomme ich nach einigen Stunden keine Daten mehr rein. Deswegen fragte ich ja nach dem websocket.
> 
> Was genau meinst du? Das hier nutzt den webhook der aktiv daten sendet. Da wird keine api abgefragt. Du solltest dein intervall erhöhen.

Komisch, wenn ich den intervall erhöhet hätte, kam das lastRing auch verspätet. Bin grad unterwegs, melde mich später dazu aber nochmal

**Comment by pietro4711** [comment]
@Xenon-s schaust Du eh im realTomeEvents-Baum nach ? Das eingestellte Intervall sollte hier ja gar keine Rolle mehr spielen...

**Comment by peterbaumert** [comment]
Einzige was sein kann ist dass der websocket zusammenbricht. Dann baut der spätestens im interval wieder neu auf. Die fehlerfälle muss ich noch abfangen.

**Comment by pietro4711** [comment]
> Schaut auf den ersten Blick sehr gut aus !! Auch lastRing mit der Doorbell klappt tadellos ! Wie soll denn das mit dem raw-Wert funktionieren ? Einfach den Text reinschreiben hat bei mir nicht geklappt.

@peterbaumert Kannst Du mir dazu was sagen ?

**Comment by peterbaumert** [comment]
> > Schaut auf den ersten Blick sehr gut aus !! Auch lastRing mit der Doorbell klappt tadellos !
> > 
> > Wie soll denn das mit dem raw-Wert funktionieren ? Einfach den Text reinschreiben hat bei mir nicht geklappt.
> > 
> > 
> 
> Ne da wird einfach genau das reingeschrieben was von der api kommt. Und das könntest du mir hier her kopieren dann kann ich direkt states aus den werten machen.

@pietro4711 Schau mal hier :)

**Comment by pietro4711** [comment]
ah alles klar:
raw:
`{
  "lcdMessage": {
    "type": "LEAVE_PACKAGE_AT_DOOR",
    "text": "LEAVE PACKAGE AT DOOR",
    "resetAt": null
  }
}
`
SomeId entspricht der ID der Doorbell

**Comment by fladdie** [comment]
Zur Vollständigkeit hier noch der Raw Wert von lastRing

`{"lastMotion":1633273487070,"isMotionDetected":true,"lastRing":1633273487435}`

**Comment by peterbaumert** [comment]
Perfekt. Danke euch für die infos. Bau das dann noch alles ein und versuch noch paar websocket fehler abzufangen.

**Comment by peterbaumert** [comment]
> Zur Vollständigkeit hier noch der Raw Wert von lastRing
> 
> 
> 
> `{"lastMotion":1633273487070,"isMotionDetected":true,"lastRing":1633273487435}`

lastRing und isMotionDetected=false heisst quasi ein Ninja hat geklingelt? :D

**Comment by pietro4711** [comment]
achja, hab übrigens folgende wiederkehrende log-meldung: 
State value to set for "unifi-protect.0.motions.lastMotion.end" has to be stringified but received type "number"

**Comment by Xenon-s** [comment]
@peterbaumert gibt's auch die Möglichkeit, dass du Ring = true übergibst, wenn es geklingelt hat? Mit dem timestamp ist das problematisch, startet der Adapter mal neu, wird der timestamp geändert und es wird dann klingeln erkannt

**Comment by peterbaumert** [comment]
@Xenon-s ja, momentan ist das noch alles "quick and dirty". Dass er am Anfang die werte zurücksetzt soll so nicht bleiben.
( bzgl "dirty" wird sich wahrscheinlich nicht viel ändern am code aber das interessiert euch ja nicht :D )

**Comment by Xenon-s** [comment]
@peterbaumert da ich selber Adapter Entwickler bin, bin ich immer froh wenn Code sauber und nicht dirty ist. Dann kann man auch schneller helfen 😉

**Comment by peterbaumert** [comment]
Was steht denn so alles im log wenn du das auf silly stellst?

**Comment by Xenon-s** [comment]
mit dem realtime event funktioniert es ;) Danke dir !

**Comment by Lucascoco** [comment]
Hallo, wo steht denn Ring=true? würde gerne eine Telegram und Alexa Message dann versenden. Danke
